src/solver/ts-generator/thermal.cpp

```diff
@@ -157,42 +166,22 @@ void GeneratorTempData::writeResultsToDisk(const Data::Area& area,
     {
         pTempFilename.reserve(study.folderOutput.size() + 256);
 
-        pTempFilename.clear() << study.folderOutput << SEP << "ts-generator" << SEP << "thermal"
-                              << SEP << "mc-" << currentYear << SEP << area.id << SEP
-                              << cluster.id() << ".txt";
+        pTempFilename.clear() << "ts-generator" << SEP << "thermal" << SEP << "mc-" << currentYear
+                              << SEP << area.id << SEP << cluster.id() << ".txt";
 
         assert(cluster.series);
         enum
         {
             precision = 0
         };
-        cluster.series->series.saveToCSVFile(pTempFilename, precision);
-    }
 
-    ++pProgression;
-}
+        std::string buffer;
+        cluster.series->series.saveToBuffer(buffer, precision);
 
-void GeneratorTempData::prepareOutputFoldersForAllAreas(uint year)
-{
-    String folder;
-    folder.reserve(study.folderOutput.size() + 256);
-
-    study.areas.each([&](Data::Area& area) {
-        if (archive)
-        {
-            folder.clear();
-            folder << study.folderOutput << SEP << "ts-generator" << SEP << "thermal" << SEP
-                   << "mc-" << year << SEP << area.id;
+        pWriter->addJob(pTempFilename.c_str(), buffer);
+    }
```

Comment:
Please remove **prepareOutputFoldersForAllAreas** declaration in class **GeneratorTempData**